src/framework/scene-grab.js

```diff
@@ -80,13 +80,22 @@ class SceneGrab {
         });
     }
 
-    allocateRenderTarget(renderTarget, device, format, isDepth, mipmaps, isDepthUniforms) {
+    resizeCondition(source, requestedRenderTarget, device) {
+        const width = source ? source.width : device.width;
+        const height = source ? source.height : device.height;
+        if (requestedRenderTarget) {
+            return width !== requestedRenderTarget.width || height !== requestedRenderTarget.height;
+        }
+        return width !== device.width || height !== device.height;
+    }
```

Comment:
I'm not sure that's correct, it should be this way I think
```
    resizeCondition(resizeTarget, cameraTarget, device) {
        const width = cameraTarget?.width || device.width;
        const height = cameraTarget?.height || device.height;
        return width !== resizeTarget?.width || height !== resizeTarget?.height;
    }
```

Reply:
Ah, didn't know you can OR together undefined and a value. Yes that would make it even terser.

Reply:
Using object destructuring it would be even shorter:

```js
function resizeCondition(resizeTarget, cameraTarget, device) {
    const {width, height} = cameraTarget || device;
    return width !== resizeTarget?.width || height !== resizeTarget?.height;
}
```

The generated ES5 code is also shorter (354 vs 488 chars ): https://babeljs.io/repl/#?browsers=&build=&builtIns=false&corejs=3.6&spec=false&loose=false&code_lz=GYVwdgxgLglg9mABAJwKYGcYC9UGEEAmMsCAFGpjgCoCGyA5qlADSIQ0C2qyNtDTrAqgBuMCKgCUiAN4AoRArYJ0URAHcYBKAAtEAXjaduvOoygB-AHQatugD53EQ0eOuadAbnmKIy1dtQYem1VA3YuHj4zKwCgkMQHJxExVEtY4KgvRRQmEGQkGx1EAEI9AwpsVCimK0L7R3T40vKMSuqLNMCMrwBfWVBIEiQKnHwwIiGAJnJW6lMBQwiTfhYkl0kZbwVfMBUZOtZGqB79ReN2hMdnFKzFNCg8gvddZpzKKvmOusvEI5Kyt5tT4xLohXqyIA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=script&lineWrap=true&presets=env&prettier=false&targets=&version=7.18.13&externalPlugins=&assumptions=%7B%7D

(untested, maybe I miss something here)